backend/src/layer/data/flickr_api/flickr_api_handler.rs

```diff
@@ -18,35 +18,74 @@ pub struct FlickrApiHandler {
 
 lazy_static! {
     static ref LONG_URL_REGEX: Regex =
-        Regex::new(r"(https://www.flickr.com/photos/)(\w+)/(\d+)([/]{0,1})")
+        Regex::new(r"(https://www\.flickr\.com/photos/)([\w@]+)(/)(\d+)(/?)")
             .expect("regex creation failed");
     static ref SHORT_URL_REGEX: Regex =
-        Regex::new(r"(https://flic.kr/p/)([\d\w]+)").expect("regex creation failed");
+        Regex::new(r"(https://flic\.kr/p/)(\w+)").expect("regex creation failed");
 }
 
+const ALPHABET: &str = "123456789abcdefghijkmnopqrstuvwxyzABCDEFGHJKLMNPQRSTUVWXYZ";
+
 impl FlickrApiHandler {
     #[must_use]
-    #[allow(clippy::missing_const_for_fn)] // cannot be made const because of compiler error`
+    #[allow(clippy::missing_const_for_fn)] // cannot be made const because of compiler error
     pub fn new(info: FlickrInfo) -> Self {
         Self {
             request: ApiRequest::new(info.api_key),
         }
     }
 
-    // URL TYPE 1: https://www.flickr.com/photos/gerdavs/52310534489/ <- remove last '/'
-    // URL TYPE 2: https://flic.kr/p/2oRguN3
+    // URL TYPE 1.1: https://www.flickr.com/photos/gerdavs/52310534489/, id group = 4
+    // URL TYPE 1.2: https://www.flickr.com/photos/198319418@N06/53077317043, id group = 4
+    // URL TYPE 2: https://flic.kr/p/2oRguN3, id group = 2
     // Both cases: Split with '/' and get last member (= photo_id).
-    fn determine_photo_id(mut url: &str) -> Result<&str> {
-        if !SHORT_URL_REGEX.is_match(url) && !LONG_URL_REGEX.is_match(url) {
-            return Err(ImageHosterError::FormatNotFound(format!(
+    fn determine_photo_id(url: &str) -> Result<String> {
+        if SHORT_URL_REGEX.is_match(url) {
+            let groups = SHORT_URL_REGEX
+                .captures(url)
+                .expect("url matches regex but captures could not be created");
```

Comment:
I could not imagine a scenario where this captures creation can fail

Reply:
you could (and should) simplify the if expression: the `captures` method returns none if the regex does not match.
`if let`would be a good fit here.